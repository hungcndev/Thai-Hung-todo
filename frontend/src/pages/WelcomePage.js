import React from "react";
import NavBar from "../components/NavBar/NavBar";
import { useNavigate } from "react-router-dom";

export default function WelcomePage() {
  const navigate = useNavigate();

  return (
    <div>
      <NavBar />
      <div className="w-full py-6 text-center">
        <div className="text-6xl font-semibold mt-16">
          Welcome to <span className="text-red">Todap</span>
        </div>
        <div className="text-6xl font-semibold mt-4">
          Let's manage your todo list
        </div>
        <div className="text-lg font-medium mt-6">
          This is an assigment from week 5 & 6 of Code Academy 2022
        </div>
        <button
            onClick={() => navigate("/signup")}
            className="font-semibold text-xl border-b border-red rounded-md py-2 px-8 mt-7 bg-red text-white"
          >
          Get Started
        </button>
      </div>
    </div>
  );
}
