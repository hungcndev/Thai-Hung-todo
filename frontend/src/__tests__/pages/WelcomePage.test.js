import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import WelcomePage from "../../pages/WelcomePage";

afterEach(cleanup);

describe("Welcome Page", () => {
  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <WelcomePage />
      </BrowserRouter>
    );
    expect(
      asFragment(
        <BrowserRouter>
          <WelcomePage />
        </BrowserRouter>
      )
    ).toMatchSnapshot();
  });
});
