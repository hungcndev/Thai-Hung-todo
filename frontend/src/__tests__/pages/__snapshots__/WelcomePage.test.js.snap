// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page should render and match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="w-full mt-16 py-6 text-center"
    >
      <div
        class="text-6xl font-semibold"
      >
        Welcome to 
        <span
          class="text-xred"
        >
          Todap
        </span>
      </div>
      <div
        class="text-6xl font-semibold mt-4"
      >
        Let's manage your todo list
      </div>
      <div
        class="text-lg font-medium mt-6 mb-12"
      >
        This is an assigment project of Code Academy 2022
      </div>
      <a
        class="font-semibold text-xl border-b border-xred hover:text-white rounded-md py-2 px-8 bg-xred hover:bg-red-500 text-white"
        href="/signup"
      >
        Get started
      </a>
    </div>
  </div>
</DocumentFragment>
`;
